src/components/MainContainer/drawers/BottomDrawer.tsx

```diff
@@ -0,0 +1,83 @@
+import { FC, useEffect, useRef, useState } from 'react'
+import { createPortal } from 'react-dom'
+import { motion, type Variants, AnimatePresence } from 'motion/react'
+import { CommonComponentProps } from '@/components/commonProps'
+import { useLayoutContext } from '../LayoutProvider'
+import { cn } from '@/lib/utils'
+import { SIDEBAR_CLOSED_WIDTH, SIDEBAR_OPENED_WIDTH } from '../sidebars/SidebarDesktop'
+import { MAIN_CONTAINER_MAX_WIDTH } from '../ContainerDesktop'
+
+const TRANSITION = {
+  duration: 0.3,
+  ease: 'circOut',
+}
+
+const VARIANTS: Variants = {
+  hidden: { y: '100%', opacity: 0 },
+  visible: { y: '0%', opacity: 1 },
+  exit: { y: '100%', opacity: 0 },
+}
+
+interface BottomActionBarBaseProps extends CommonComponentProps {
+  portalContainer?: HTMLElement | null
+}
+
+export const BottomDrawer: FC<BottomActionBarBaseProps> = ({ className = '', portalContainer }) => {
+  const { isSidebarOpen, isDrawerOpen, drawerContent } = useLayoutContext()
+  const sidebarWidth = isSidebarOpen ? SIDEBAR_OPENED_WIDTH : SIDEBAR_CLOSED_WIDTH
+
+  const drawerRef = useRef<HTMLDivElement>(null)
+  const [drawerHeight, setDrawerHeight] = useState(0)
+
+  useEffect(() => {
+    if (drawerRef.current) {
+      setDrawerHeight(drawerRef.current.offsetHeight)
+    } else {
+      setDrawerHeight(0)
+    }
+  }, [drawerContent])
+
+  useEffect(() => {
+    const container = document.querySelector('#main-container') as HTMLElement | null
+    if (!container) return
+
+    if (isDrawerOpen && drawerHeight > 0) {
+      container.style.paddingBottom = `${drawerHeight}px`
+    } else {
+      container.style.paddingBottom = '0px'
+    }
+
+    return () => {
+      if (container) container.style.paddingBottom = '0px'
+    }
+  }, [drawerHeight, isDrawerOpen])
+
+  return createPortal(
```

Comment:
Here, instead of using `createPortal`, I could use the portal from [radix](https://www.radix-ui.com/primitives/docs/utilities/portal)

Reply:
For now it's ok to use the react-dom one 👍🏼